### PR TITLE
Always make a corresponding Phoenix account.

### PR DESCRIPTION
**Pull request by DFurnes**
#### What's this PR do?
A lot of functionality still relies on a Phoenix account existing for every user. Now that we've shipped Northstar authentication, we don't need every Phoenix account to have a password in Phoenix which simplifies things a lot!

Alongside DoSomething/phoenix#7229, this should ensure all Northstar accounts will have a Drupal ID.

#### How should this be reviewed?
This shouldn't cause any issues, but we should do some thorough testing just in case!

~Relies on DoSomething/phoenix#7229.~ Merged! ✅ 
References #406.

#### Checklist
- [x] Documentation added for changed endpoints.
- [ ] Tests added for new features/bug fixes.
- [ ] Post a message in #api if this includes something that causes a rebuild!  

---
For review: @angaither @weerd

### CODE REVIEW

**Comment by DFurnes** [comment]
[Always create a Phoenix account when registering users via Northstar API endpoints.](https://trello.com/c/7zCUSYma/145-always-create-a-phoenix-account-when-registering-users-via-northstar-api-endpoints)

File: app/Auth/Registrar.php [hunk 1]
```diff
@@ -185,31 +186,37 @@ public function register($input, $user = null)
...
     {
-        // @TODO: we can't create a Drupal user without an email. Do we just create an @mobile one like we had done previously?
         try {
-            $drupal_id = $this->phoenix->register($user, $password);
+            $drupal_id = $this->phoenix->createDrupalUser($user);
             $user->drupal_id = $drupal_id;
-        } catch (\GuzzleHttp\Exception\ClientException $e) {
+        } catch (ClientException $e) {
             // If user already exists (403 Forbidden), try to find the user to get the UID.
             if ($e->getCode() == 403) {
```
Comment (weerd) [inline app/Auth/Registrar.php]: triple equals?

Comment (DFurnes) [inline app/Auth/Registrar.php]: Ah! Updated in 32a7773609f66e9242e569179471aaaa896cfeeb.

File: app/Services/Phoenix.php [hunk 2]
```diff
@@ -108,18 +109,21 @@ public function campaigns($id = null)
...
      */
-    public function register($user, $password)
+    public function createDrupalUser($user)
     {
         $payload = $user->toArray();
 
         // Format user object for consumption by Drupal API.
         $payload['birthdate'] = format_date($user->birthdate, 'Y-m-d');
         $payload['user_registration_source'] = $user->source;
-        $payload['password'] = $password;
```
Comment (angaither) [inline app/Services/Phoenix.php]: maybe I'm missing this elsewhere, but does it still get included if it is set?

Comment (DFurnes) [inline app/Services/Phoenix.php]: Nope - to simplify things, we're never sending password out of Northstar now. The only case where a password would also be stored within Drupal is if a user registers via the legacy Phoenix registration modal.

**Comment by DFurnes** [comment]
[Unplanned Work: Fix issues caused by Celebs Gone Good users not having Drupal IDs.](https://trello.com/c/kIvMHGsr/202-unplanned-work-fix-issues-caused-by-celebs-gone-good-users-not-having-drupal-ids)

**Pull request merged**